implementations/rust/ockam/ockam_api/src/kafka/outlet_service/prefix_forwarder.rs

```diff
@@ -0,0 +1,107 @@
+use ockam_core::{Address, AllowAll, AllowOnwardAddress};
+
+/// This service applies a prefix to the provided static forwarding address.
+/// This service was created mainly to keep full compatibility with the existing
+/// erlang implementation.
+pub struct PrefixForwarderService {
+    prefix: String,
+    secure_channel_listener_flow_control_id: FlowControlId,
+}
+impl PrefixForwarderService {
+    pub async fn create(
+        context: &Context,
+        secure_channel_listener_flow_control_id: FlowControlId,
+    ) -> Result<()> {
+        // add the this worker as consumer for the secure channel listener
+        let worker_address = Address::from_string(KAFKA_OUTLET_CONSUMERS);
+        context.flow_controls().add_consumer(
+            worker_address.clone(),
+            &secure_channel_listener_flow_control_id,
+            FlowControlPolicy::SpawnerAllowMultipleMessages,
+        );
+
+        let worker = Self {
+            prefix: "consumer_".to_string(),
+            secure_channel_listener_flow_control_id,
+        };
+
+        context
+            .start_worker_with_access_control(
+                worker_address,
+                worker,
+                AllowAll,
+                AllowOnwardAddress(DefaultAddress::FORWARDING_SERVICE.into()),
+            )
+            .await
+    }
+}
+
+#[ockam::worker]
+impl Worker for PrefixForwarderService {
+    type Message = Vec<u8>;
+    type Context = Context;
+
+    async fn handle_message(
+        &mut self,
+        ctx: &mut Context,
+        msg: Routed<Self::Message>,
+    ) -> Result<()> {
+        // the payload is just a string
+        let address = match msg.payload().get(1..) {
+            Some(address) => match from_utf8(address) {
+                Ok(v) => v.to_string(),
+                Err(_e) => {
+                    return Err(ockam_core::Error::new(
+                        Origin::Application,
+                        Kind::Invalid,
+                        "invalid address",
+                    ));
+                }
+            },
+            None => {
+                return Err(ockam_core::Error::new(
+                    Origin::Application,
+                    Kind::Invalid,
+                    "invalid address",
+                ));
+            }
+        };
+
+        let new_address = format!("{}_{}", &self.prefix, address);
+
+        debug!(
+            "prefix forwarder, renamed from {} to {}",
+            address, new_address
+        );
+
+        let mut bytes = new_address.clone().into_bytes();
+        let mut new_payload: Vec<u8> = vec![bytes.len() as u8];
+        new_payload.append(&mut bytes);
+
+        let mut message = msg.into_local_message();
+        let transport_message = message.transport_mut();
+
+        transport_message.onward_route.step()?;
+
+        // prefix consumer_ to the address
+        transport_message.payload = new_payload;
+
+        ctx.forward(message).await?;
```

Comment:
That's questionable

Reply:
What do you mean? the worker is spawned with `AllowOnwardAddress(DefaultAddress::FORWARDING_SERVICE.into()),`

Reply:
It's a redundancy at best, possible vulnerability at worst